public/app/features/alerting/unified/state/actions.ts

```diff
@@ -456,6 +459,32 @@ export const saveRuleFormAction = createAsyncThunk(
             const rulerConfig = getDataSourceRulerConfig(thunkAPI.getState, GRAFANA_RULES_SOURCE_NAME);
             const rulerClient = getRulerClient(rulerConfig);
             identifier = await rulerClient.saveGrafanaRule(values, evaluateEvery, existing);
+
+            // track if the user switched from simplified routing to policies routing or vice versa
+            if (isGrafanaRulerRule(existing?.rule)) {
```

Comment:
nit: can we have a separate function here like `reportSwitchingRoutingType(...)`?